internal/provider/gateway_name.go

```diff
@@ -101,3 +106,23 @@ func syncContractsNameGateways(d *schema.ResourceData, gw *workloads.GatewayName
 	d.SetId(fmt.Sprint(gw.ContractID))
 	return
 }
+
+// validateBackends ensures that if tlsPassthrough is set, the backend form is ip:port
+// and if not set, the form is http://ip[:port]
+func validateBackends(backnends []zos.Backend, tlsPassthrough bool) error {
+	for _, u := range backnends {
+		if tlsPassthrough {
+			if err := u.Valid(tlsPassthrough); err != nil {
+				return errors.Wrap(err, "backend with tls passthrough should be in the form ip:port")
+			}
+
+			continue
+		}
+
+		if err := u.Valid(tlsPassthrough); err != nil {
+			return errors.Wrap(err, "backends without tls passthrough should be in the form http://ip[:port]")
+		}
+	}
```

Comment:
I don't think they need to be separated, both calls are identical

Reply:
true, the error message is different though.

Reply:
an error suitable for both like this is fine :)
https://github.com/threefoldtech/terraform-provider-grid/pull/741/files#diff-19baefa0fb8c49a0ec14d248eb79f8998a1fe7cacceadec7f922c234c977dc1eR73